src/assignment/views/AssignmentDetail.tsx

```diff
@@ -114,7 +114,8 @@ const AssignmentDetail: FC<DefaultSecureRouteProps<{ id: string }>> = ({
 	const isContributor = assignment?.share_type === ShareWithColleagueTypeEnum.GEDEELD_MET_MIJ;
 	const isSharedWithOthers =
 		assignment?.share_type === ShareWithColleagueTypeEnum.GEDEELD_MET_ANDERE;
-	const isBeingEdited = editStatuses && !!editStatuses[id];
+	const isBeingEdited =
+		editStatuses && (!!editStatuses[id] || !(editStatuses[id].subjectId === user?.profile?.id));
```

Comment:
when you campare 2 optional properties, always make sure at least one is filled in, otherwise you could be comparing 
```
undefined === undefined => true
```

also you want editingUserId and not subjectId, which contains the assignmentId or collectionId

```suggestion
		editStatuses && (!!editStatuses[id] || !(!!editStatuses[id]?.editingUserId && editStatuses[id]?.editingUserId === user?.profile?.id));
```